src/csharp/Grpc.Tools/build/_protobuf/Google.Protobuf.Tools.targets

```diff
@@ -115,27 +116,33 @@
           DependsOnTargets="Protobuf_Compile"
           Condition=" '$(Language)' == 'C#' " />
 
-  <Target Name="_Protobuf_SelectFiles">
+  <Target Name="_Protobuf_SetProtoRoot">
     <!-- Guess .proto root for the files. Whenever the root is set for a file explicitly,
          leave it as is. Otherwise, for files under the project directory, set the root
          to "." for the project's directory, as it is the current when compiling; for the
-         files outside of project directory, use each .proto file's directory as the root. -->
+         files outside of project directory, use each .proto file's directory as the root 
+		 or Protobuf_ProtoRoot if set. -->
     <FindUnderPath Path="$(MSBuildProjectDirectory)"
                    Files="@(Protobuf->WithMetadataValue('ProtoRoot',''))">
       <Output TaskParameter="InPath" ItemName="_Protobuf_NoRootInProject"/>
       <Output TaskParameter="OutOfPath" ItemName="_Protobuf_NoRootElsewhere"/>
     </FindUnderPath>
+	
+    <ItemGroup>
+      <Protobuf>
```

Comment:
can you please explain how come this update the existing protobuf items (which seems to be the intention).
I might be missing something obvious (I'm not an msbuild expert).